src/Umbraco.Core/Routing/UmbracoRequestPaths.cs

```diff
@@ -37,7 +37,7 @@ public UmbracoRequestPaths(IOptions<GlobalSettings> globalSettings, IHostingEnvi
                 .EnsureStartsWith('/').TrimStart(_appPath.EnsureStartsWith('/')).EnsureStartsWith('/');
 
             _mvcArea = globalSettings.Value.GetUmbracoMvcArea(hostingEnvironment);
-            _defaultUmbPaths = new List<string> { "/" + _mvcArea, "/" + _mvcArea + "/" };
+            _defaultUmbPaths = new HashSet<string>(StringComparer.InvariantCultureIgnoreCase) { "/" + _mvcArea, "/" + _mvcArea + "/" };
```

Comment:
It would be worth looking at https://youtu.be/__7V6H-MMWg?t=699 for some further optimisation. StringComparer.InvariantCultureIgnoreCase calculates a new hash for every .Contains().